holoviews/util/__init__.py

```diff
@@ -929,9 +929,29 @@ def _get_streams(self, map_obj, watch=True):
         of supplied stream classes and instances are processed and
         added to the list.
         """
+        if isinstance(self.p.streams, dict):
+            streams = defaultdict(dict)
+            stream_specs, params = [], {}
+            for name, p in self.p.streams.items():
+                if not isinstance(p, param.Parameter):
+                    raise ValueError("Stream dictionary must map operation keywords "
+                                     "to parameter names. Cannot handle %r type."
+                                     % type(stream))
+                if inspect.isclass(p.owner) and issubclass(p.owner, Stream):
+                    if p.name != name:
+                        streams[p.owner][p.name] = name
+                    else:
+                        streams[p.owner] = {}
+                else:
+                    params[name] = p
+            stream_specs = streams_list_from_dict(params)
+            stream_specs += [stream(rename=rename) for stream, rename in streams.items()]
+        else:
+            stream_specs = self.p.streams
```

Comment:
This is my improved suggestion. If you want move this part out into a utility. I prefer this approach because it doesn't needlessly wrap stream parameters into `Params` streams but rather simply creates the stream instance the parameter is referring to and also avoids creating one stream per parameter by grouping by the parameter owner.

Reply:
I do prefer this. It relies on the `p.owner` being the correct stream class which you can then pass `rename` to (and I like the use of default dict here). This avoids the dangling stream instances that were created only to pass their parameters to `Params`. Thanks!

Reply:
No need to pull out a utility as I don't think it would make much sense out of context. Instead I've added a few clarifying comments in 324871a97